walkthroughs/howto-external-traffic/infrastructure/ecs-service.yaml

```diff
@@ -0,0 +1,320 @@
+    Properties:
+      RequiresCompatibilities:
+        - 'FARGATE'
+      Family: 'gateway'
+      NetworkMode: 'awsvpc'
+      Cpu: 256
+      Memory: 512
+      TaskRoleArn:
+        'Fn::ImportValue': !Sub "${EnvironmentName}:TaskIamRoleArn"
+      ExecutionRoleArn:
+        'Fn::ImportValue': !Sub "${EnvironmentName}:TaskExecutionIamRoleArn"
+      ContainerDefinitions:
+        - Name: envoy
+          Image: !Ref EnvoyImage
+          Essential: true
+          Ulimits:
+            - Name: "nofile"
+              HardLimit: 15000
+              SoftLimit: 15000
+          PortMappings:
+            - ContainerPort: 9901
+              Protocol: 'tcp'
+            - ContainerPort: 15000
+              Protocol: 'tcp'
+            - ContainerPort: 15001
+              Protocol: 'tcp'
+            - ContainerPort: 80
+              Protocol: 'tcp'
+          HealthCheck:
+            Command:
+              - 'CMD-SHELL'
+              - 'curl -s http://localhost:9901/server_info | grep state | grep -q LIVE'
+            Interval: 5
+            Timeout: 2
+            Retries: 3
+          LogConfiguration:
+            LogDriver: 'awslogs'
+            Options:
+              awslogs-group:
+                Fn::ImportValue: !Sub "${EnvironmentName}:ECSServiceLogGroup"
+              awslogs-region: !Ref AWS::Region
+              awslogs-stream-prefix: 'gateway-envoy'
+          Environment:
+            - Name: 'APPMESH_RESOURCE_ARN'
+              Value: !Sub 'mesh/${AppMeshMeshName}/virtualGateway/ColorGateway'
+            - Name: 'ENVOY_LOG_LEVEL'
+              Value: 'debug'
+
+  ColorGatewayService:
+    Type: 'AWS::ECS::Service'
+    DependsOn:
+      - WebLoadBalancerRule
+    Properties:
+      Cluster:
+        'Fn::ImportValue': !Sub "${EnvironmentName}:ECSCluster"
+      DeploymentConfiguration:
+        MaximumPercent: 200
+        MinimumHealthyPercent: 100
+      DesiredCount: 1
+      LaunchType: FARGATE
+      ServiceRegistries:
+        - RegistryArn:
+            'Fn::GetAtt': ColorGatewayServiceDiscoveryRecord.Arn
+      NetworkConfiguration:
+        AwsvpcConfiguration:
+          AssignPublicIp: DISABLED
+          SecurityGroups:
+            - 'Fn::ImportValue': !Sub "${EnvironmentName}:ECSServiceSecurityGroup"
+          Subnets: 
+            - 'Fn::ImportValue': !Sub "${EnvironmentName}:PrivateSubnet1"
+            - 'Fn::ImportValue': !Sub "${EnvironmentName}:PrivateSubnet2"
+      TaskDefinition: { Ref: ColorGatewayTaskDefinition }
+      HealthCheckGracePeriodSeconds: 300
+      LoadBalancers:
+        - ContainerName: envoy
+          ContainerPort: 80
+          TargetGroupArn: !Ref WebTargetGroup
+
+  PublicLoadBalancerSG:
+    Type: AWS::EC2::SecurityGroup
+    Properties:
+      GroupDescription: Access to the public facing load balancer
+      VpcId:
+        'Fn::ImportValue': !Sub "${EnvironmentName}:VPC"
+      SecurityGroupIngress:
+          - CidrIp: 0.0.0.0/0
+            IpProtocol: -1
```

Comment:
Can you restrict the port list to those port numbers only used within this walkthrough?

Reply:
https://github.com/aws/aws-app-mesh-examples/pull/501/files

as per that PR
```
          IpProtocol: tcp
          FromPort: 80
          ToPort: 80
```